### PR TITLE
Major overhaul for Monitoring docs: Part 1

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

The Monitoring pages need some overhauls to focus on the following:

- MinIO publishes metrics using the Prometheus Data Model
- Generalize for any service which can scrape Prometheus data
- Add tutorials for other services which support Prometheus data

Not included in this pass:

- Refresh of Server and Audit Logs
- Audit log to kafka
- End-to-ends for Containerized environments

### CODE REVIEW

File: source/operations/monitoring/metrics-and-alerts.rst [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+.. metric:: minio_node_disk_used_bytes
+
+   Total storage used on a disk. The ``server`` and ``disk`` label identifies the host machine and disk respectively.
+
+Lifecycle Management Metrics
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. metric:: minio_cluster_ilm_transitioned_bytes
+
+   Total amount of data transitioned using :ref:`tiering/transition lifecycle management rules <minio-lifecycle-management-tiering>`
```
Comment (djwfyi) [inline source/operations/monitoring/metrics-and-alerts.rst:192]: Should this be `Total number of bytes of data` ... to match up with how we talk about other metrics above?

File: source/operations/monitoring/metrics-and-alerts.rst [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+
+   Total number of bytes of S3 traffic sent per bucket
+
+.. metric:: minio_bucket_traffic_received_bytes
+
+   Total number of bytes of S3 traffic received per bucket
+
+.. metric:: minio_s3_requests_error_total
+
+   Total number S3 requests with errors
```
Comment (djwfyi) [inline source/operations/monitoring/metrics-and-alerts.rst:299]: Total number **of** S3 requests...

File: source/operations/monitoring/metrics-and-alerts.rst [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+
+.. metric:: minio_s3_requests_inflight_total
+
+   Total number of S3 requests currently in flight
+
+.. metric:: minio_s3_requests_total
+
+   Total number S3 requests
```
Comment (djwfyi) [inline source/operations/monitoring/metrics-and-alerts.rst:307]: Total number **of** S3 requests

File: source/operations/monitoring/metrics-and-alerts.rst [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+
+   Limit on total number of open file descriptors for the MinIO Server process.
+
+.. metric:: minio_node_file_descriptor_open_total
+
+   Total number of open file descriptors by the MinIO Server process.
+
+.. metric:: minio_node_io_rchar_bytes
+
+   Total bytes read by the process from the underlying storage system including
```
Comment (djwfyi) [inline source/operations/monitoring/metrics-and-alerts.rst:354]: Total **number of** bytes ...

Comment (djwfyi) [reply source/operations/monitoring/metrics-and-alerts.rst:354]: All metrics in this section need this for consistency with other metrics.

**Review by djwfyi — APPROVED** [review]
Some consistency and wording suggestions, but otherwise this all looks good to me.

File: source/operations/monitoring.rst [hunk 5]
```diff
@@ -12,22 +12,27 @@ Metrics and Alerts
...
-You can download these metrics as either ``.png`` images or ``.csv`` files.
+   * - :ref:`minio-metrics-collect-using-prometheus`
+     - Configure Prometheus to Monitor and Alert for a MinIO deployment
+
+       Configure MinIO to query the Prometheus deployment to enable historical metrics via the MinIO Console
+
+   * - :ref:`minio-metrics-influxdb`
+     - Configure InfluxDB to Monitor and Alert for a MinIO deployment.
+
+Other metrics and analytics software suites which support the Prometheus data format may work regardless of their inclusion on the above list.
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/operations/monitoring.rst:35]: DOne!

File: source/operations/monitoring/metrics-and-alerts.rst [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. metric:: minio_bucket_objects_size_distribution
+
+   Distribution of object sizes in the bucket, includes label for the bucket 
+   name.
+
+.. metric:: minio_bucket_usage_object_total
+
+   Total number of objects
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/operations/monitoring/metrics-and-alerts.rst:92]: I should not have done periods in the first place, these are barely complete sentences as is :D

File: source/operations/monitoring/metrics-and-alerts.rst [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+These metrics are only populated for MinIO clusters with 
+:ref:`minio-bucket-replication-serverside` enabled.
+
+.. metric:: minio_bucket_replication_failed_bytes
+
+   Total number of bytes failed at least once to replicate.
+
+.. metric:: minio_bucket_replication_pending_bytes
+
+   Total bytes pending to replicate.
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/operations/monitoring/metrics-and-alerts.rst:110]: Will keep it for now, can twiddle with specifics later

File: source/operations/monitoring/metrics-and-alerts.rst [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+
+   Total free usable capacity online in the cluster.
+
+.. metric:: minio_cluster_capacity_usable_total_bytes
+
+   Total usable capacity online in the cluster.
+
+.. metric:: minio_node_disk_free_bytes
+
+   Total storage available on a disk. The ``server`` and ``disk`` label identifies the host machine and disk respectively.
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/operations/monitoring/metrics-and-alerts.rst:177]: Actually re-wrote this to be a little more useful for users who are not familiar with the data model.

File: source/operations/monitoring/metrics-and-alerts.rst (hunk 1 above)
Comment (ravindk89) [reply source/operations/monitoring/metrics-and-alerts.rst:192]: Total number of bytes is probably fine...here `of data` would definitely be redundant.

File: source/operations/monitoring/metrics-and-alerts.rst (hunk 4 above)
Comment (ravindk89) [reply source/operations/monitoring/metrics-and-alerts.rst:354]: Done

**Pull request merged**